feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/domain/cloudBackup/enterPassword/EnterCloudBackupInteractor.kt

```diff
@@ -68,6 +69,30 @@ class RealEnterCloudBackupInteractor(
 
     override suspend fun restoreCloudBackupPassword(password: String): Result<Unit> {
         return cloudBackupService.fetchAndDecryptExistingBackup(password)
-            .map { cloudBackupService.session.setSavedPassword(password) }
+            .mergeBackupAndLocalState(password)
+            .onSuccess { cloudBackupService.session.setSavedPassword(password) }
+    }
```

Comment:
It's a potential problem here. We can get destructive diff error that will block password changing flow.

I think if we have diffing issue we should continue password changing since we anyway will see destructive diff in sync screen